gkeys/gkeys/actions.py

```diff
@@ -474,7 +485,29 @@ def speccheck(self, args):
                     sdata = convert_pf(pub_pass, ['pub', 'sign', 'final'])
                     sdata = convert_yn(sdata, ['auth', 'encrypt'])
                     self.output('', SPECCHECK_SUMMARY % sdata)
-
+                    '''Email reminder code'''
+                    if args.email in ['expiry']:
+                        uid = ''
+                        if gkey.uid:
+                            uid = gkey.uid[0]
+                            match = re.findall(r'[\w\.-]+@[\w\.-]+', uid)
+                            uid = ''
+                            if match:
+                                uid = _unicode(match[0])
```

Comment:
Also, this code should be put as a utility function somewhere.  Then it is re-usable for other actions or reminders...